src/inference/hmc.jl

```diff
@@ -557,40 +426,12 @@ gen_metric(dim::Int, spl::Sampler{<:Hamiltonian}) = AHMC.UnitEuclideanMetric(dim
 gen_metric(dim::Int, ::AHMC.UnitPreconditioner)   = AHMC.UnitEuclideanMetric(dim)
```

Comment:
Does this belong in AHMC?

Reply:
I will double check.

Reply:
@cpfiffer I think some of them that are not dependent on `spl` can simply use https://github.com/TuringLang/AdvancedHMC.jl/blob/master/src/adaptation/precond.jl#L357